packages/sanity/src/presentation/preview/OpenPreviewButton.tsx

```diff
@@ -1,20 +1,33 @@
 import {LaunchIcon} from '@sanity/icons'
+import {urlSearchParamPreviewPerspective} from '@sanity/preview-url-secret/constants'
 import {Text} from '@sanity/ui'
-import {useCallback} from 'react'
+import {useCallback, useMemo} from 'react'
 import {useTranslation} from 'sanity'
 
 import {Button, Tooltip} from '../../ui-components'
 import {presentationLocaleNamespace} from '../i18n'
+import {type PresentationPerspective} from '../types'
+import {encodeStudioPerspective} from '../util/encodeStudioPerspective'
 import {type PreviewProps} from './Preview'
 
 /** @internal */
 export function OpenPreviewButton(
   props: Pick<PreviewProps, 'openPopup'> & {
-    previewLocationOrigin: string
+    previewLocationOrigin?: string
     previewLocationRoute: string
+    perspective: PresentationPerspective
+    targetOrigin: string
   },
 ): React.ReactNode {
-  const {openPopup, previewLocationOrigin, previewLocationRoute} = props
+  const {openPopup, previewLocationOrigin, previewLocationRoute, perspective, targetOrigin} = props
+
+  const openPreviewLink = useMemo(() => {
+    const url = new URL(previewLocationRoute, previewLocationOrigin || targetOrigin)
+    url.searchParams.set(urlSearchParamPreviewPerspective, encodeStudioPerspective(perspective))
```

Comment:
Yep this is fine. The `withoutSecretParams` util was originally designed to remove `sanity-preview-secret` and `sanity-preview-path`.
When we added `sanity-preview-perspective` it was added to the same util just to make the location URL cleaner, since the search param is noisy and an implementation detail. It would be better if we originally made sure to strip `sanity-preview-perspective` in the places we didn't want to display it, instead of stripping it everywhere by default and then have to add it back in where needed 🙌